command/scroll/on_edit.go

```diff
@@ -14,19 +14,31 @@ type Commander interface {
 	Execute(bind.Bindable)
 }
 
-type OnMove struct {
+type OnEdit struct {
 	Commander Commander
 }
 
-func (o *OnMove) Name() string {
-	return "scroll-on-caret-movement"
+func (o *OnEdit) Name() string {
+	return "scroll-movement-on-edit"
 }
 
-func (o *OnMove) OpName() string {
-	return "caret-movement"
+func (o *OnEdit) OpName() string {
+	return "input-handler"
 }
 
-func (o *OnMove) Moved(e input.Editor, carets []int) {
-	s := o.Commander.Bindable("scroll").(*Scroller)
-	o.Commander.Execute(s.For(ToRune(carets[0])))
+func (o *OnEdit) Applied(e input.Editor, edits []input.Edit) {
+	// TODO: handling multiple editing
+	if len(edits) == 1 {
+		edit := edits[0]
+		s := o.Commander.Bindable("scroll").(*Scroller)
+		if len(edit.New) == 0 {
+			if len(edit.Old) == 1 {
+				o.Commander.Execute(s.For(ToRune(edit.At, Up)))
+			} else {
+				o.Commander.Execute(s.For(ToOldOffset()))
+			}
+		} else if len(edit.Old) == 0 {
+			o.Commander.Execute(s.For(ToRune(edit.At+len(edit.New), NoDirection)))
+		}
+	}
```

Comment:
Did the `OnMove` not work correctly?  I tested a few different scenarios and I thought it was working (the caret moves whenever text is edited, so it _should_ scroll when you edit text too).  I thought it was more comprehensive to scroll whenever the caret moves, not just when the text changes.

Reply:
I'm going to merge this, but I think that we should edit it so that _most_ scrolling is done during caret movement.  I can see the situations where we need to scroll during edits, too, but most of the time, editing the code will cause the caret to move.

I think the caret might be getting moved by something other than the caret movement op, which is not _supposed_ to happen, but if so it would skip the scrolling hook.  What I'm working on now should make those problems more obvious, so that we can ensure that all caret movement is triggered by the caret movement op.